language-support/java/codegen/src/ledger-tests/scala/com/digitalasset/TestUtil.scala

```diff
@@ -19,55 +18,63 @@ import com.daml.ledger.javaapi.data
 import com.daml.ledger.javaapi.data._
 import com.daml.ledger.resources.ResourceContext
 import com.daml.platform.apiserver.SeedService.Seeding
-import com.daml.platform.apiserver.services.GrpcClientResource
 import com.daml.platform.common.LedgerIdMode
-import com.daml.platform.sandbox
-import com.daml.platform.sandbox.SandboxServer
+import com.daml.platform.sandbox.config.SandboxConfig
+import com.daml.platform.sandboxnext.SandboxNextFixture
 import com.daml.platform.services.time.TimeProviderType
 import com.daml.ports.Port
 import com.google.protobuf.Empty
 import io.grpc.Channel
-import org.scalatest.Assertion
+import org.scalatest.{Assertion, Suite}
 
 import scala.concurrent.Future
 import scala.jdk.CollectionConverters._
 import scala.language.implicitConversions
 
-object TestUtil {
+trait SandboxFixture extends SandboxNextFixture {
+  self: Suite =>
 
-  def testDalf =
+  protected val damlPackages: List[File] = List(
     new File(BazelRunfiles.rlocation("language-support/java/codegen/ledger-tests-model.dar"))
-
-  val LedgerID = "ledger-test"
+  )
+  protected val ledgerIdMode: LedgerIdMode =
+    LedgerIdMode.Static(LedgerId(TestUtil.LedgerID))
+  protected override def config: SandboxConfig = SandboxConfig.defaultConfig.copy(
+    port = Port.Dynamic,
+    ledgerIdMode = ledgerIdMode,
+    damlPackages = damlPackages,
+    timeProviderType = Some(TimeProviderType.Static),
+    engineMode = SandboxConfig.EngineMode.Dev,
+    seeding = Some(Seeding.Weak),
+  )
 
   def withClient(
       testCode: Channel => Assertion
   )(implicit resourceContext: ResourceContext): Future[Assertion] = {
-    val config = sandbox.DefaultConfig.copy(
-      seeding = Some(Seeding.Weak),
-      port = Port.Dynamic,
-      damlPackages = List(testDalf),
-      ledgerIdMode = LedgerIdMode.Static(LedgerId(LedgerID)),
-      timeProviderType = Some(TimeProviderType.WallClock),
-    )
-
-    val channelOwner = for {
-      server <- SandboxServer.owner(config)
-      channel <- GrpcClientResource.owner(server.port)
-    } yield channel
-    channelOwner.use(channel => Future(testCode(channel))(resourceContext.executionContext))
+    Future(testCode(channel))(resourceContext.executionContext)
   }
+}
+
+object TestUtil {
+
+  val LedgerID = "ledger-test"
 
   // unfortunately this is needed to help with passing functions to rxjava methods like Flowable#map
   implicit def func2rxfunc[A, B](f: A => B): io.reactivex.functions.Function[A, B] = f(_)
   private def randomId = UUID.randomUUID().toString
 
+  def uniqueId(): String = UUID.randomUUID.toString
+
+  protected[daml] def getUniqueParty(name: String): String = s"${name}_${uniqueId()}"
```

Comment:
```suggestion
  private[daml] def getUniqueParty(name: String): String = s"${name}_${uniqueId()}"
```

Reply:
rather than trying to make up unique parties can you allocate them via the party admin service on the ledger API? On Canton that will be required anyway so this seems like a step in the right direction.

Reply:
> rather than trying to make up unique parties can you allocate them via the party admin service on the ledger API

done 👍

Reply:
good stuff, thank you!